src/Definition.php

```diff
@@ -145,7 +145,15 @@ public function getMaker()
      */
     public function setCallback(callable $callback)
     {
-        $this->callback = $callback;
+        if ($this->callback) { // "stack" the callbacks so that any previous callbacks will also be called.
+            $oldCallback = $this->callback;
+            $this->callback = function ($model, $saved) use ($oldCallback, $callback) {
+                // If the old callback returns false, the whole thing should return false.
+                return $oldCallback($model, $saved) !== false && $callback($model, $saved) !== false;
```

Comment:
Returning false to exit the callback chain is incompatible with the fact that false also determines if a save should go ahead or not. What if someone wants to exit the callback chain but still wants another persit to run?

Reply:
If I understand correctly, the current implementation will not persist the model if the callback returns false? So this line should make it so that if any of the callbacks return false, the model won't be persisted. Would you rather that all callbacks are called regardless of whether a prior one "failed" and then return false if any of them failed afterwards to prevent it being persisted?

Reply:
> If I understand correctly, the current implementation will not persist the model if the callback returns false?

No, it will not persist the model **again**. Possibly someone might want the model to be updated in the database again after their callback does something, but does not want the remaining callbacks to be executed.

Reply:
Ah, okay.
So I should change it to something like
```php
$this->callback = function ($model, $saved) use ($oldCallback, $callback) {
        $oldResult = $oldCallback($model, $saved);
        $newResult = $callback($model, $saved);
        return $oldResult !== false && $newResult !== false;
};
```
So that all the callbacks are always called, and returning false in any of them will prevent the second persist.

Then if we move this to `addCallback()`, then you can still use `setCallback()` to override all the other callbacks if you don't want them called.